modules/tinymce/src/plugins/imagetools/test/ts/browser/ImageToolsDisabledButtonsTest.ts

```diff
@@ -0,0 +1,74 @@
+import { Mouse, UiFinder } from '@ephox/agar';
+import { before, describe, it } from '@ephox/bedrock-client';
+import { McEditor, TinyDom } from '@ephox/mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import ImagePlugin from 'tinymce/plugins/image/Plugin';
+import ImageToolsPlugin from 'tinymce/plugins/imagetools/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('browser.tinymce.plugins.imagetools.ImageToolsDisabledButtonsTest', () => {
+
+  const settings = {
+    plugins: 'image imagetools',
+    toolbar: 'editimage fliph flipv rotateleft rotateright',
+    base_url: '/project/tinymce/js/tinymce',
+    height: 900
+  };
+
+  const editImageButtonSelector = '[role="toolbar"] button[title="Edit image"]';
+  const flipHorizontallyImageButtonSelector = '[role="toolbar"] button[title="Flip horizontally"]';
+  const flipVerticallyImageButtonSelector = '[role="toolbar"] button[title="Flip vertically"]';
+  const rotateCounterClockwiseImageButtonSelector = '[role="toolbar"] button[title="Rotate counterclockwise"]';
+  const rotateClockwiseImageButtonSelector = '[role="toolbar"] button[title="Rotate clockwise"]';
+
+  before(() => {
+    Theme();
+    ImageToolsPlugin();
+    ImagePlugin();
+  });
+
+  const assertButtonsStatus = (editor: Editor, disabled: boolean) => {
+    const container = TinyDom.container(editor);
+    UiFinder.exists(container, `${editImageButtonSelector}[aria-disabled="${disabled}"]`);
+    UiFinder.exists(container, `${flipHorizontallyImageButtonSelector}[aria-disabled="${disabled}"]`);
+    UiFinder.exists(container, `${flipVerticallyImageButtonSelector}[aria-disabled="${disabled}"]`);
+    UiFinder.exists(container, `${rotateCounterClockwiseImageButtonSelector}[aria-disabled="${disabled}"]`);
+    UiFinder.exists(container, `${rotateClockwiseImageButtonSelector}[aria-disabled="${disabled}"]`);
+  };
+
+  const insertRemoteImage = (editor: Editor) => {
+    const remoteImageSrc = 'https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png';
+    editor.setContent(`<p><img src="${remoteImageSrc}" width="50" height="50" /></p>`);
+  };
+
+  it('TINY-7772: Edit, flip and rotate should be disabled for remote images without imagetools_proxy set', async () => {
+    const editor = await McEditor.pFromSettings<Editor>(settings);
+    insertRemoteImage(editor);
```

Comment:
This could be done in a `beforeEach` as it's setup logic.

Reply:
There's one test that uses different `settings`:
`const editor = await McEditor.pFromSettings<Editor>({ ...settings, imagetools_proxy: 'foo.php' });`

If they had the same settings how I'd remove the `editor` at the end of each test?

Reply:
I didn't comment on that line? I commented on the part about setting the initial content so it contains the remote image.

Reply:
Oh wait I see what you mean. You're right we don't have an editor instance at that point so we can't. In that case ignore this sorry.